product_internal_reference_code/models/res_company.py

```diff
@@ -2,13 +2,57 @@
 # Copyright 2017 Quartile Limited
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import models, fields
+from odoo import models, fields, api
 
 
 class ResCompany(models.Model):
     _inherit = "res.company"
 
     internal_code_prefix = fields.Char(
         string="Internal Reference Prefix",
-        copy=True,
+        copy=False,
+        require=True,
+        help="A prefix code which is used to automatically generate the "
+             "internal reference when creating a new product."
     )
+
+    @api.multi
+    def create(self, vals):
+        res = super(ResCompany, self).create(vals)
+        # Create the sequence for the company upon creation
+        if 'internal_code_prefix' in vals:
+            prefix = vals['internal_code_prefix']
+        seq = {
+            'name': res.name + ' - Product Sequence',
+            'code': 'product.product.internal_code',
+            'prefix': prefix or '',
+            'padding': 5,
+            'number_increment': 1,
+            'company_id': res.id,
+        }
+        self.env['ir.sequence'].create(seq)
+        return res
```

Comment:
Not sure if we want this method.  If we keep this, set `res.id` to prefix.  User can later change the value in ir.sequence view.

Reply:
Lets keep this method to ensure there will be always an ir.sequence for each company